src/tests/tools/iotc_test_echoserver.cc

```diff
@@ -0,0 +1,164 @@
+/* Copyright 2018-2019 Google LLC
+ *
+ * This is part of the Google Cloud IoT Device SDK for Embedded C,
+ * it is licensed under the BSD 3-Clause license; you may not use this file
+ * except in compliance with the License.
+ *
+ * You may obtain a copy of the License at:
+ *  https://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "iotc_test_echoserver.h"
+
+#include <netinet/in.h>
+#include <string.h>
+#include <thread>
+
+namespace iotctest {
+
+EchoTestServer::EchoTestServer(uint16_t socket_type, uint16_t port,
+                               uint16_t protocol_type, const char* host)
+    : socket_type_(socket_type), test_port_(port),
+      protocol_type_(protocol_type), host_(host) {
+  CreateServer();
```

Comment:
https://google.github.io/styleguide/cppguide.html#Doing_Work_in_Constructors
https://abseil.io/tips/42

I'd suggest the following:
````
class EchoTestServer {
 public:
  static std::unique_ptr<EchoTestServer> Create(uint16_t socket_type, uint16_t protocol_type, 
                                                                              std::string host, uint16_t port);
 private:
  EchoTestServer(...);
  CreateServer();
};
```
and `Create` calls the private constructor + `CreateServer`.

Reply:
Also note the change in parameter order and `std::string` for host.